dropwizard-health/src/main/java/io/dropwizard/health/response/DetailedJsonHealthResponseProvider.java

```diff
@@ -0,0 +1,82 @@
+package io.dropwizard.health.response;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.health.HealthStateAggregator;
+import io.dropwizard.health.HealthStateView;
+import io.dropwizard.health.HealthStatusChecker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.ws.rs.core.MediaType;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class DetailedJsonHealthResponseProvider implements HealthResponseProvider {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DetailedJsonHealthResponseProvider.class);
+
+    @Nonnull
+    private final HealthStatusChecker healthStatusChecker;
+    @Nonnull
+    private final HealthStateAggregator healthStateAggregator;
+    @Nonnull
+    private final ObjectMapper mapper;
+
+    public DetailedJsonHealthResponseProvider(@Nonnull final HealthStatusChecker healthStatusChecker,
+                                              @Nonnull final HealthStateAggregator healthStateAggregator,
+                                              @Nonnull final ObjectMapper mapper) {
+        this.healthStatusChecker = Objects.requireNonNull(healthStatusChecker);
+        this.healthStateAggregator = Objects.requireNonNull(healthStateAggregator);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Nonnull
+    @Override
+    public HealthResponse fullHealthResponse(@Nullable String type) {
+        final Collection<HealthStateView> healthStateViews = healthStateAggregator.healthStateViews();
+        final String responseBody;
+        try {
+            responseBody = mapper.writeValueAsString(healthStateViews);
+        } catch (final Exception e) {
+            LOGGER.error("Failed to serialize health state views: {}", healthStateViews, e);
+            throw new RuntimeException(e);
+        }
+        final boolean healthy = healthStatusChecker.isHealthy(type);
+
+        return new HealthResponse(healthy, responseBody, MediaType.APPLICATION_JSON);
+    }
+
+    @Nonnull
+    @Override
+    public HealthResponse minimalHealthResponse(@Nullable String type) {
+        final boolean healthy = healthStatusChecker.isHealthy(type);
+
+        return new HealthResponse(healthy, null, MediaType.APPLICATION_JSON);
+    }
+
+    @Nonnull
+    @Override
+    public HealthResponse partialHealthResponse(@Nullable final String type, @Nonnull final Collection<String> names) {
```

Comment:
What do you think about having sub-classes for `PartialHealthResponse` and `MinimalHealthResponse` to better differentiate them. `PartialHealthResponse` could have a `healthStateViews` member and perhaps even some kind of validation for the view names so that a request for a health check name that is not defined in the config will be detected and a warning can be shown.

Reply:
I made `HealthResponse` abstract, and added two sub-classes `DetailedHealthResponse` and `MinimalHealthResponse`, to make the distinctions between the two clearer, and hopefully make the code a little less error-prone.